spec/features/admin/products_spec.rb

```diff
@@ -470,16 +470,96 @@
       expect("#{uri.path}?#{uri.query}").to eq spree.admin_product_images_path(product, filter)
     end
 
-    scenario "editing a product's variant unit scale", js: true do
-      product = create(:simple_product, name: 'a product', supplier: @supplier2)
+    context "editing a product's variant unit scale", js: true do
+      let!(:product) { create(:simple_product, name: 'a product', supplier: @supplier2) }
 
-      visit spree.edit_admin_product_path product
-      select 'Weight (kg)', from: 'product_variant_unit_with_scale'
-      click_button 'Update'
-      expect(flash_message).to eq('Product "a product" has been successfully updated!')
-      product.reload
-      expect(product.variant_unit).to eq('weight')
-      expect(product.variant_unit_scale).to eq(1000)
+      # assertions commented out refer to bug:
+      # https://github.com/openfoodfoundation/openfoodnetwork/issues/7180
+
+      before do
+        Spree::Config.available_units = "g,lb,oz,kg,T,mL,L,kL"
+        visit spree.edit_admin_product_path product
+      end
+
+      it 'as weight (g)' do
+        select 'Weight (g)', from: 'product_variant_unit_with_scale'
+        click_button 'Update'
+        expect(flash_message).to eq('Product "a product" has been successfully updated!')
+        product.reload
+        expect(product.variant_unit).to eq('weight')
+        # expect(page).to have_select('product_variant_unit_with_scale', :selected => 'Weight (g)')
+        expect(product.variant_unit_scale).to eq(1)
+      end
```

Comment:
This code has been duplicated a few times. This can be simplified with shared examples:
https://relishapp.com/rspec/rspec-core/docs/example-groups/shared-examples

Reply:
Thank you so much for the really thorough and helpful review @mkllnk. I hope to have addressed the other ones but especially the shared_examples on the latest commit.